apps/angular/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,73 @@
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { Task } from './model/task';
+import { taskActions } from './store/tasks/task.action';
+import { getTaskListSelector } from './store/tasks/task.selector';
+import { LoaderService } from './ui/loader/loader.service';
+import { TaskComponent } from './ui/task/task.component';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [
+    CommonModule,
+    MatProgressSpinnerModule,
+    MatProgressBarModule,
+    MatSnackBarModule,
+    TaskComponent,
+  ],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
+    <div class="relative flex h-max justify-center">
+      @if (loaderService.isLoading$()) {
+        <div class="fixed z-10 w-full">
+          <mat-progress-bar mode="indeterminate" value="40"></mat-progress-bar>
+        </div>
+        <!-- <div class="absolute w-full h-full z-10 bg-slate-600 opacity-70">
+          <mat-spinner class="top-1/2 left-1/2 z-20"></mat-spinner>
+        </div> -->
+      }
+      <div class="m-32">
+        <div class="mb-5 text-center text-3xl">Angular CRUD Challenge</div>
+        <div class="flex justify-between gap-5">
+          <div class="w-30vw mb-5 text-center text-xl">Title</div>
+          <div class="mb-5 text-center text-xl">Actions</div>
+        </div>
+        @for (todo of todos; track $index) {
+          <app-task [task]="todo"></app-task>
+          <hr />
+        }
+      </div>
     </div>
   `,
   styles: [],
 })
-export class AppComponent implements OnInit {
-  todos!: any[];
-
-  constructor(private http: HttpClient) {}
+export class AppComponent implements OnInit, OnDestroy {
+  private stodos = signal<Task[]>([]);
+  private sysSubs: Subscription = new Subscription();
+  public get todos(): Task[] {
+    return this.stodos();
+  }
+  public loaderService = inject(LoaderService);
+  private store = inject(Store);
 
-  ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
+  ngOnDestroy(): void {
+    this.sysSubs.unsubscribe();
   }
 
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
+  ngOnInit(): void {
+    this.store.dispatch(taskActions.loadTasksAction());
+    this.sysSubs.add(
+      this.store.select(getTaskListSelector).subscribe({
```

Comment:
don't code imperatively. 

you have selectSignal in Ngrx, so you can immedialy do stodo = selectSIgnal(mySelector). 

Avoid setting class properties inside ngOnInit.

Reply:
Great!